azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/Deployment.java

```diff
@@ -0,0 +1,277 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.domain;
+
+import static com.google.common.collect.ImmutableList.copyOf;
+
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableMap;
+import org.jclouds.domain.JsonBall;
+import org.jclouds.javax.annotation.Nullable;
+
+import com.google.auto.value.AutoValue;
+import org.jclouds.json.SerializedNames;
+
+@AutoValue
+public abstract class Deployment {
+
+   public enum ProvisioningState {
+      ACCEPTED("Accepted"),
+      READY("Ready"),
+      CANCELED("Canceled"),
+      FAILED("Failed"),
+      DELETED("Deleted"),
+      SUCCEEDED("Succeeded"),
+      RUNNING("Running"),
+      UNRECOGNIZED("");
+
+      private final String key;
+
+      ProvisioningState(final String key) {
+         this.key = key;
+      }
+
+      public static ProvisioningState fromString(final String text) {
+         if (text != null) {
+            for (ProvisioningState state : ProvisioningState.values()) {
+               if (text.equalsIgnoreCase(state.key)) {
+                  return state;
+               }
+            }
+         }
+         return UNRECOGNIZED;
+      }
+   }
+
+   public enum DeploymentMode {
+      INCREMENTAL("Incremental"),
+      COMPLETE("Complete"),
+      UNRECOGNIZED("");
+
+      private final String key;
+
+      DeploymentMode(final String key) { this.key = key; }
+
+      public static DeploymentMode fromString(final String text) {
+         if (text != null) {
+            for (DeploymentMode mode : DeploymentMode.values()) {
+               if (text.equalsIgnoreCase(mode.key)) {
+                  return mode;
+               }
+            }
+         }
+         return UNRECOGNIZED;
+      }
+   }
+
+   @AutoValue
+   public abstract static class TypeValue {
+      public abstract String type();
+
+      public abstract String value();
+
+      @SerializedNames({"type", "value"})
+      public static TypeValue create(final String type, final String value) {
+         return new AutoValue_Deployment_TypeValue(type, value);
+      }
+   }
+
+   @AutoValue
+   public abstract static class ProviderResourceType {
+      @Nullable
+      public abstract String resourceType();
+
+      @Nullable
+      public abstract List<String> locations();
+
+      @Nullable
+      public abstract List<String> apiVersions();
+
+      @Nullable
+      public abstract Map<String, JsonBall> properties();
```

Comment:
Is there a need for collections to be nullable? Is this the same case than the `tags` properties?